build kmod with static pie

`kmod` gets flagged by `checksec`:
```
RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH    Symbols      	FORTIFY	Fortified	Fortifiable   Filename
Partial RELRO   No canary found   NX enabled    No PIE          N/A        N/A          No Symbols	N/A	0		0		/.bottlerocket/rootfs/usr/bin/kmod
```

This was missed during the conversion of `kmod` to a static binary in https://github.com/bottlerocket-os/bottlerocket/pull/3981.

Now the check with `checksec` panics:

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/slimm609/checksec/pkg/checksec.FunctionsFromSymbolTable(0xc0001720f8)
        /home/agarrcia/.local/go/path/pkg/mod/github.com/slimm609/checksec@v0.0.0-20250322005620-31f2c9784742/pkg/checksec/symbols.go:90 +0xb4f
github.com/slimm609/checksec/pkg/checksec.Canary({0x7ffc07229b5c?, 0x4?})
        /home/agarrcia/.local/go/path/pkg/mod/github.com/slimm609/checksec@v0.0.0-20250322005620-31f2c9784742/pkg/checksec/canary.go:55 +0x1e6
github.com/slimm609/checksec/pkg/utils.RunFileChecks({0x7ffc07229b5c, 0x4}, {0x0, 0x0})
        /home/agarrcia/.local/go/path/pkg/mod/github.com/slimm609/checksec@v0.0.0-20250322005620-31f2c9784742/pkg/utils/checks.go:14 +0x79
github.com/slimm609/checksec/cmd.init.func3(0xc0001e0800?, {0xc000026370?, 0x4?, 0x6113eb?})
        /home/agarrcia/.local/go/path/pkg/mod/github.com/slimm609/checksec@v0.0.0-20250322005620-31f2c9784742/cmd/file.go:18 +0x5c
github.com/spf13/cobra.(*Command).execute(0x7b7f80, {0xc000026340, 0x1, 0x1})
        /home/agarrcia/.local/go/path/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0xa7b
github.com/spf13/cobra.(*Command).ExecuteC(0x7b9000)
        /home/agarrcia/.local/go/path/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x40c
github.com/spf13/cobra.(*Command).Execute(...)
        /home/agarrcia/.local/go/path/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/slimm609/checksec/cmd.Execute()
        /home/agarrcia/.local/go/path/pkg/mod/github.com/slimm609/checksec@v0.0.0-20250322005620-31f2c9784742/cmd/root.go:31 +0xbc
main.main()
        /home/agarrcia/.local/go/path/pkg/mod/github.com/slimm609/checksec@v0.0.0-20250322005620-31f2c9784742/main.go:13 +0x10c
```

--- a/packages/kmod/0001-meson-add-support-for-static-builds.patch
+++ b/packages/kmod/0001-meson-add-support-for-static-builds.patch
@@ -18,7 +18,7 @@ index a0cf675..4b16d6a 100644
  
 +kmod_link_args = []
 +if get_option('default_library') == 'static'
-+  kmod_link_args += ['-static']
++  kmod_link_args += ['-static-pie']
 +endif
 +
  kmod = executable(

